common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -140,6 +148,63 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+        final String methodTag = TAG + ":onResume";
+        final Bundle extras = getExtras();
+        Logger.info(methodTag, "extras: " + extras.keySet());
+        if (getSwitchBrowserCoordinator().isExpectingSwitchBrowserResume()) {
+            resumeSwitchBrowser(extras);
+        }
+    }
+
+    /**
+     * Get the extras from the activity intent.
+     *
+     * @return Bundle with the extras
+     */
+    @NonNull
+    private Bundle getExtras() {
+        final Activity activity = getActivity();
+        if (activity == null) {
+            return Bundle.EMPTY;
+        }
+        final Intent intent = activity.getIntent();
+        if (intent == null) {
+            return Bundle.EMPTY;
+        }
+        final Bundle extras =  intent.getExtras();
+        return extras == null ? Bundle.EMPTY : extras;
+    }
+
+    /**
+     * Resume the switch browser protocol flow.
+     *
+     * @param extras Bundle with the data to resume the switch browser protocol flow.
+     */
+    private void resumeSwitchBrowser(@NonNull final Bundle extras) {
+        final String methodTag = TAG + ":resumeSwitchBrowser";
+        try {
+            Logger.info(methodTag, "Resuming switch browser flow");
+            getSwitchBrowserCoordinator().processSwitchBrowserResume(
+                    extras,
+                    mRedirectUri,
+                    mClientId,
+                    (switchBrowserResumeUri, switchBrowserResumeHeaders) -> {
+                        mAuthorizationRequestUrl = switchBrowserResumeUri.toString();
```

Comment:
>mAuthorizationRequestUrl [](http://example.com/codeflow?start=24&length=24)

I think we should not updating mAuthorizationRequestUrl rather use it directly with webview. maybe modify launchWebView() to launchWebView(url..)

Reply:
eca86720aafb090be96d3fad49d875ccaa4ddc0e